src/hierarchy/Animal.ts

```diff
@@ -63,4 +68,15 @@ export default abstract class Animal implements ICanWalk {
   getCanHaveFamily(): boolean {
     return this._canHaveFamily;
   }
+  public giveBirth():AnimalCandidate[] | void{
+    if (this.isMale) {
+      throw Error(
+        `Only Females can give birth and ${this.name} is a proud male ${this.constructor.name}`
+      );
+    } else if (this.home === undefined) {
+      throw Error(
+        `${this.name} can't give birth because she doesn't have a home to meet a male`
```

Comment:
i didnt quite get what you mean... What doesn't work

Reply:
if you start the app and open the console in the browser you will se an uncaught error "Simba can't give birth because she doesn't have a home to meet a male". This crashes the app and the rest of the code that's meant for testing in the app.ts doesn't run

Reply:
i thought it wasn't supposed to run, just to throw an error

Reply:
if you want to stop the app from working once an animal that can't give birth tries to that's fine but just for testing purposes wrap the code that throws an error in a try catch so the rest of the testing code in app.ts runs. See in app.ts file how Konstantin wrapped some parts of the code in try catch.

Reply:
OK, will do!